### DISCUSSION TITLE
ETA tasks limit

### DISCUSSION THREAD

**Discussion by akunafin** [celery/celery#9086] [Q&A]
We use rabbitmq as a broker. There is a queue with millions of ETA tasks. When I start a single worker it will eventually go out of RAM because there is no prefetch limit. I found that 10 replicas can handle all ETA tasks if I start them at the same time. However, if one of the workers dies, it releases all consumed messages and rabbit pushes these messages to other workers resulting in out of memory and another restarts.
I tried to set prefetch_multiplier, but it turns out that it has no effect on ETA tasks. Is there any other way I can limit the number of ETA tasks that a single worker can schedule?

**Comment by thedrow** [comment]
No, there's no such mechanism currently.
We'd consider a PR to fix this issue.
